yunwoo/programmers-1/문자열-내-p와-y의-개수.js

```diff
@@ -0,0 +1,19 @@
+function solution(s) {
+  let answer = true;
+  const str = s.toUpperCase();
+  const stringArray = str.split("");
+  const pCount = stringArray.filter((item) => "P" === item).length;
+  const yCount = stringArray.filter((item) => "Y" === item).length;
+
+  if (pCount !== yCount) answer = false;
+
+  return answer;
+}
```

Comment:
```suggestion
function solution(s) {
  const lowerString = s.toLowerCase();
  let pCount = 0;
  let yCount = 0;

  for (let el of lowerString) {
    if (el === "p") pCount++;
    else if (el === "y") yCount++;
  }
  return pCount === yCount;
}
```
- 윤우님 엄청 일찍 제출해주셨네요 ㅋㅋㅋ 감사합니다 또 굉장히 잘 풀어주셨네요!! 👍 그래도 약간의 효율성면을 고려해서 의견을 드리자면 `filter` 배열 메서드는 `O(N)`의 시간복잡도를 갖고있습니다.
- 시간복잡도 O(N)이라하면 그냥 for문으로 배열 전체를 1번 순회했다고 생각하시면됩니다.
- 따라서 현재 윤우님 코드를 보면 filter가 2번 실행하고 `O(2N)`, 그 전에 문자열을 split까지해서 배열로 변환해야하는 작업이 효율면에서 조금 아쉽다고 느껴졌어요! 🥲

<br />

- 수정 코드는 제가 풀이 한 코드인데 한번 참고만 해보세요 ! 단 한번의 for문으로 풀이한 코드입니다 `O(N)`
- + answer라는 변수가 명확하고 직관적이긴한데 저는 그냥 pCount === yCount를 비교한 값을 곧바로 리턴해봤습니다 이건 코드 스타일 차이라 참고하셔도 좋고, 그냥 진행하셔도 좋을 것 같아요!